runtime/src/iree-amd-aie/aie_runtime/iree_aie_router.cc

```diff
@@ -394,6 +424,15 @@ std::optional<std::map<PathEndPoint, SwitchSettings>> Router::findPaths(
     }
   }
 
+  // group flows based on packetGroupId
+  std::map<int, std::vector<Flow>> groupedFlows;
+  for (auto &f : impl->flows) {
+    if (groupedFlows.count(f.packetGroupId) == 0) {
+      groupedFlows[f.packetGroupId] = std::vector<Flow>();
+    }
+    groupedFlows[f.packetGroupId].push_back(f);
+  }
```

Comment:
```suggestion
  std::map<int, std::vector<Flow>> groupedFlows;
  for (auto &f : impl->flows) {
    groupedFlows[f.packetGroupId].push_back(f);
  }
```

just accessing std::map with a key creates the value

https://godbolt.org/z/bTPv1e61G